### PR TITLE
Remove p4c patch, p4c already fixed

### PR DESCRIPTION
**Pull request by kamleshbhalui**
p4c dpdk backend require some passes from bmv2

Signed-off-by: Kamlesh Kumar <kamleshbhalui@gmail.com>

### CODE REVIEW

File: build/networking/scripts/build_p4c.sh
Comment (kamleshbhalui) [inline build/networking/scripts/build_p4c.sh:44]: dpdk backend depends on some bmv2 passes, so we can not exclude from build

File: build/networking_ebpf/Dockerfile.ubuntu
Comment (kamleshbhalui) [inline build/networking_ebpf/Dockerfile.ubuntu:72]: require to fix atleast on ubuntu 18.04
: program not found or is not executable
--grpc_out: protoc-gen-grpc: Plugin failed with status code 1.

Comment (mestery) [reply build/networking_ebpf/Dockerfile.ubuntu:72]: Nice catch!

File: build/networking/scripts/build_p4c.sh
Comment (mestery) [reply build/networking/scripts/build_p4c.sh:44]: We should honestly just stop disabling anything here and build with everything enabled.

**Review by mestery — APPROVED** [review]
LGTM assuming this passes CI.

**Comment by mestery** [comment]
@kamleshbhalui Did you want to test out the container image here before I merge this? I'm fine either way, let me know.

File: build/networking/scripts/build_p4c.sh
Comment (mestery) [inline build/networking/scripts/build_p4c.sh:17]: I think you need to update this as well or we'll be using a really ancient version of p4c during the build.

**Review by mestery — CHANGES_REQUESTED** [review]

File: build/networking/scripts/build_p4c.sh
Comment (kamleshbhalui) [reply build/networking/scripts/build_p4c.sh]: done

File: build/networking/scripts/build_p4c.sh
Comment (kamleshbhalui) [reply build/networking/scripts/build_p4c.sh:44]: done

Comment (stolsma) [reply build/networking/scripts/build_p4c.sh:44]: I'm not sure that the dpdk backend depends on for bmv2 compiled code. i.e. the dpdk backend uses BMV2 passes (code) but should compile that code when creating the p4c-dpdk binary, it should not depend on code compiled for the p4c-bmv2 binary. If that is the case then we should create a issue in the p4c repo!

**Review by stolsma — COMMENTED** [review]
Stumbled into not compiled p4c also (see #140) Will try if this PR will solve that....

**Comment by kamleshbhalui** [comment]
> Stumbled into not compiled p4c also (see #140) Will try if this PR will solve that....

yes it will solve that.

**Comment by kamleshbhalui** [comment]
>>I'm not sure that the dpdk backend depends on for bmv2 compiled code. i.e. the dpdk backend uses BMV2 passes (code) but should compile that code when creating the p4c-dpdk binary, it should not depend on code compiled for the p4c-bmv2 binary. If that is the case then we should create a issue in the p4c repo!

The passes from bmv2 which we use in dpdk , happens to depend to one bmv2 specific IR Node, which won't get generated if we disable the bmv2 from build. 
I think we have to internalize the pass to dpdk rather depending on bmv2.(I will work upon it.)

**Review by mestery — APPROVED** [review]
Thanks for the fixes!

**Comment by stolsma** [comment]
This will work without any problems to create p4c-dpdk. Don't know what you see is needed from bmv2???

```
cmake -DENABLE_BMV2=OFF -DENABLE_EBPF=OFF -DENABLE_UBPF=OFF -DENABLE_P4C_GRAPHS=OFF -DENABLE_P4TEST=OFF -DENABLE_GTESTS=OFF ..
```

**Comment by mestery** [comment]
@stolsma I'm not sure why we would disable everything else, why not just build all of p4c?

**Comment by stolsma** [comment]
Because it takes time and space for one, plus a problem in one of those backends will disturb the building of the p4-ovs/dpdk image. As is the problem discovered in #140 !

**Comment by mestery** [comment]
The eventual goal should be to use all of the backends. I believe the real, actual problem here is that we should not have to compile p4c but should use a packaged version instead, and this should be taken up in the p4c GitHub.

**Comment by stolsma** [comment]
? Why would the p4-ovs/dpdk image have the possibility to compile the ebpf p4 or even the bmv2 code?

**Comment by mestery** [comment]
My point is IPDK will eventually use all those backends. We already have a p4-ebpf target merged, for example. The only reason we're even having to compile p4c is because it's all so cutting edge at this point. The goal should be to get to a point where we don't have to compile it but can instead use packages.

**Pull request merged**
